### PR TITLE
Feature: "Not any pests" chat filter and "No pests!" title

### PR DESCRIPTION
**Pull request by saga-00**
## What
Adds a filter which hides the "not any pests" message. Also adds a "No Pests!" title, which triggers when you use the Pest Tracker with no pests on your garden. Those features can be used separately.

<details>
<summary>Images</summary>

image
java_YM4kzF8yW4

</details>

## Changelog New Features
+ Added No Pests Chat Filter. - saga
    * Removed the chat message "There are no Pests on your Garden!".
+ Added No Pests Title. - saga
    * Shows a title when you use the Pest Tracker without any pests to clear.

### CODE REVIEW

File: src/main/java/at/hannibal2/skyhanni/features/chat/ChatFilter.kt [hunk 1]
```diff
@@ -403,6 +403,10 @@ class ChatFilter {
         "§d[\\w']+ the Fairy§r§f: Have a great life!".toPattern()
     )
 
+    private val gardenPestMessage = listOf(
+        "§cThere are not any Pests on your Garden right now! Keep farming!"
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/chat/ChatFilter.kt]: could we make this a repo pattern now so it doesnt have to be done later

File: src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt [hunk 2]
```diff
@@ -154,6 +156,15 @@ object PestFinder {
 
     private var lastKeyPress = SimpleTimeMark.farPast()
 
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!config.noPestTitle) return
+
+        val containsCheck = GardenAPI.inGarden() && event.message == "§cThere are not any Pests on your Garden right now! Keep farming!"
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt]: Same as other comment

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
The message should be made into a repo pattern, probably in PestFinder and then ChatFilter can use that pattern

File: src/main/java/at/hannibal2/skyhanni/features/chat/ChatFilter.kt [hunk 3]
```diff
@@ -489,6 +490,7 @@ class ChatFilter {
         dungeonConfig.soloClass && DungeonAPI.inDungeon() && message.isPresent("solo_class") -> "solo_class"
         dungeonConfig.soloStats && DungeonAPI.inDungeon() && message.isPresent("solo_stats") -> "solo_stats"
         dungeonConfig.fairy && DungeonAPI.inDungeon() && message.isPresent("fairy") -> "fairy"
+        config.gardenNoPest && GardenAPI.inGarden() && PestFinder.noPestsChatPattern.matches(message) -> "gardenPest"
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/chat/ChatFilter.kt]:
```suggestion
        config.gardenNoPest && GardenAPI.inGarden() && PestFinder.noPestsChatPattern.matches(message) -> "garden_pest"
```

File: src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt [hunk 4]
```diff
@@ -154,6 +162,15 @@ object PestFinder {
 
     private var lastKeyPress = SimpleTimeMark.farPast()
 
+    @SubscribeEvent
+    fun onChat(event: LorenzChatEvent) {
+        if (!config.noPestTitle) return
+
+        val containsCheck = GardenAPI.inGarden() && noPestsChatPattern.matches(event.message)
+
+        if (event.blockedReason == "gardenPest" || containsCheck) LorenzUtils.sendTitle("§eNo pests!", 2.seconds)
```
Comment (CalMWolfs) [inline src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt]: im like 99% sure that even if the message is blocked this event will still recieve it and match the pattern, therefore there is no need to check the blocked reason, also probably return if not on garden before checking if the config option is enabled

**Review by CalMWolfs — CHANGES_REQUESTED** [review]
changes look good, just two more things i saw

File: src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt (hunk 4 above)
Comment (saga-00) [reply src/main/java/at/hannibal2/skyhanni/features/garden/pests/PestFinder.kt]: indeed correct, didn't think about it, my first thought was the blockedReason path. Thank you for the changes, will be pushing again.

**Review by CalMWolfs — APPROVED** [review]

**Pull request merged**